### PR TITLE
New module - BBMap clumpify 

### PR DESCRIPTION
**Pull request by tamuanand**
## PR checklist
- [x] This comment contains a description of changes (with reason).
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag bbmap/clumpify --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag bbmap/clumpify --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by tamuanand** [comment]
This was created using guidelines given by @mahesh-panchal [here](https://github.com/nf-core/modules/pull/1971#issuecomment-1227151017)

**Comment by tamuanand** [comment]
@drpatelh @mahesh-panchal 
-requesting review and merge

File: modules/bbmap/clumpify/main.nf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    tuple val(meta), path('*.log')     , emit: log
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def raw      = meta.single_end ? "in=${reads[0]}" : "in1=${reads[0]} in2=${reads[1]}"
```
Comment (mahesh-panchal) [inline modules/bbmap/clumpify/main.nf:24]:
```suggestion
    def raw      = meta.single_end ? "in=$reads" : "in1=${reads[0]} in2=${reads[1]}"
```
This works in both cases, when `reads` is just a path, or when `reads` is an array of one path.

File: modules/bbmap/clumpify/main.nf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def clumped  = meta.single_end ? "out=${prefix}.clumped.fastq.gz" : "out1=${prefix}_1.clumped.fastq.gz out2=${prefix}_2.clumped.fastq.gz"
+    """
+    clumpify.sh \\
+        $raw \\
+        $clumped \\
+        $args \\
+        &> ${prefix}.clumpify.log
```
Comment (mahesh-panchal) [inline modules/bbmap/clumpify/main.nf:31]: Is this a single threaded program? No `task.cpus` usable?

File: modules/bbmap/clumpify/main.nf [hunk 3]
```diff
@@ -0,0 +1,37 @@
+process BBMAP_CLUMPIFY {
+    tag "$meta.id"
+    label 'process_high_memory'
```
Comment (mahesh-panchal) [inline modules/bbmap/clumpify/main.nf:3]:
```suggestion
    label 'process_single'
    label 'process_high_memory'
```
You should make sure there are settings for both `time` and `cpus` too.

**Review by mahesh-panchal — CHANGES_REQUESTED** [review]
Awesome work. Just a few minor changes required.

**Comment by mahesh-panchal** [comment]
The singularity test failing here doesn't look like a simple thing to solve.

**Comment by tamuanand** [comment]
@mahesh-panchal Thanks for all the suggestions and improvements. I have accepted and committed them

Please LMK if I missed anything

File: modules/bbmap/clumpify/main.nf (hunk 2 above)
Comment (tamuanand) [reply modules/bbmap/clumpify/main.nf:32]: This is a java based program - I just followed how other bbmap modules were written - https://github.com/nf-core/modules/blob/master/modules/bbmap/bbduk/main.nf

Comment (mahesh-panchal) [reply modules/bbmap/clumpify/main.nf:32]: I checked over the options. This one appears to be single threaded.

**Review by mahesh-panchal — APPROVED** [review]
LGTM

**Comment by tamuanand** [comment]
@mahesh-panchal - what are next steps here? Does it need additonal approvers?

Thanks

**Comment by tamuanand** [comment]
@JoseEspinosa @matthdsm - could you have a look and approve/merge this 

Thanks

**Review by matthdsm — APPROVED** [review]

**Comment by mahesh-panchal** [comment]
I was going to say nothing needed and you can enjoy merging your first content, but I see it's been merged already. Nice work.

**Comment by tamuanand** [comment]
Thanks @mahesh-panchal and @matthdsm

**Pull request merged**